### ISSUE TITLE
[java] AvoidFieldNameMatchingMethodName should consider enum class

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3936]
**Affects PMD Version:6.42.0** 

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#avoidfieldnamematchingmethodname>

**Description:**
Hi, I found a false negative about the rule AvoidFieldNameMatchingMethodName. PMD should have reported a warning in line 5. I think this bug is caused by enum class here. Thanks for your consideration.

**Code Sample demonstrating the issue:**
```java
public class C {
  enum E {
      R;
      int bar;
      void bar() {}  // should report a warning in this line
  }
}
```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Scrsloota** [comment]
Please assign this to me! 😊

**Comment by Scrsloota** [comment]
Please assign this to me! 😊
